src/main/java/io/spaship/operator/service/k8s/CommandExecutionService.java

```diff
@@ -0,0 +1,128 @@
+        if (pod.isEmpty()) {
+            throw new RuntimeException("No pod found for label " + podLabel);
+        }
+
+        // TODO :scope of improvement: add support for multiple commands
+        var command  = symbolicLinkCommand(sourceTargetTuple.getItem1(), sourceTargetTuple.getItem2());
+
+        pod.forEach(p -> {
+            try {
+                executeCommandInContainer(p, command);
+            } catch (CommandExecutionException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+   // todo :must do: 1. check both source and destination exists 2. Validate and sanitize the
+   //  command
+    private String[] symbolicLinkCommand(String source, String target) {
+        ReUsableItems.checkNull(source,target);
+        source = (BASE_HTTP_DIR.concat("/").concat(source)).toLowerCase();
+        target = (BASE_HTTP_DIR.concat("/").concat(target)).toLowerCase();
+        LOG.debug("creating a symlink of source {} to {}",source,target);
+        LOG.debug("command to be executed [ln] [-s] [{}] [{}]",source,target);
+        return new String[]{"ln", "-s", source, target};
+    }
+
+    private  void executeCommandInContainer(Pod httpdPod, String[] command) throws CommandExecutionException {
+        ReUsableItems.checkNull(httpdPod,command);
+        CountDownLatch latch = new CountDownLatch(1);
+        try (ExecWatch ignored = ocClient.pods().inNamespace(httpdPod.getMetadata().getNamespace())
+                .withName(httpdPod.getMetadata().getName())
+                .inContainer(CONTAINER_NAME).readingInput(System.in). //TODO replace the deprecated method
+                // with the new method
+                writingOutput(System.out).writingError(System.err).withTTY()
+                .usingListener(new ExecListener() {
+                    @Override
+                    public void onOpen() {
+                        LOG.debug("Executing command in container");
+                    }
+
+                    @Override
+                    public void onClose(int code, String reason) {
+                        LOG.debug("closing the listener");
+                        latch.countDown();
+                    }
+
+                    @SneakyThrows
+                    @Override
+                    public void onFailure(Throwable t, Response failureResponse) {
+                        LOG.error("Failed to execute command in container due to {}",failureResponse.body());
+                        latch.countDown();
+                    }
+
+                    @Override
+                    public void onExit(int code, Status status) {
+                        LOG.error("Command executed in container code {} reason {}",code,status);
+                        latch.countDown();
+                    }
+                }).exec(command)) {
+            latch.await();
+        } catch (Exception e) {
+            throw new CommandExecutionException("Error while executing command in container", e);
```

Comment:
Enhancement: In case of exception a proper message should be communicated to the user (via API response/SSE events). To be discussed.